src/NexusMods.App.UI/Pages/Downloads/InProgressView.axaml.cs

```diff
@@ -108,26 +126,60 @@ public InProgressView()
                     }
                 })
                 .DisposeWith(d);
+            
+            // Dynamically Update Completed Download Count
+            this.WhenAnyValue(view => view.ViewModel!.CompletedDownloadCount)
+                .OnUI()
+                .Subscribe(count =>
+                {
+                    CompletedTitleCountTextBlock.Text = StringFormatters.ToDownloadsInProgressTitle(count);
+                })
+                .DisposeWith(d);
+            
+            // Hide Completed Section if no downloads
+            this.WhenAnyValue(view => view.ViewModel!.CompletedDownloadCount)
+                .Select(count => count > 0)
+                .BindToView(this, view => view.CompletedSectionGrid.IsVisible)
+                .DisposeWith(d);
 
-            // Bind Selected Items
+            // Bind inProgress Selected Items
             Observable.FromEventPattern<SelectionChangedEventArgs>(
-                    addHandler => ModsDataGrid.SelectionChanged += addHandler,
-                    removeHandler => ModsDataGrid.SelectionChanged -= removeHandler)
+                    addHandler => InprogressDataGrid.SelectionChanged += addHandler,
+                    removeHandler => InprogressDataGrid.SelectionChanged -= removeHandler)
                 .Do(_ =>
                 {
-                    ViewModel!.SelectedTasks.Edit(updater =>
+                    ViewModel!.SelectedInProgressTasks.Edit(updater =>
                     {
                         updater.Clear();
-                        foreach (var item in ModsDataGrid.SelectedItems)
+                        foreach (var item in InprogressDataGrid.SelectedItems)
+                        {
+                            if (item is IDownloadTaskViewModel task)
+                                updater.Add(task);
```

Comment:
Can items be anything else? If not, this should have a `Debug.Assert`.

Reply:
I think it can, I think headers and such.